src/flows/components/SecretsList.tsx

```diff
@@ -0,0 +1,166 @@
+import React, {
+  FC,
+  useCallback,
+  useContext,
+  useEffect,
+  useState,
+  useMemo,
+} from 'react'
+import {connect, ConnectedProps, useDispatch, useSelector} from 'react-redux'
+
+// Components
+import {
+  Button,
+  ComponentColor,
+  ComponentSize,
+  FlexBox,
+  IconFont,
+  JustifyContent,
+} from '@influxdata/clockface'
+import SearchWidget from 'src/shared/components/search_widget/SearchWidget'
+import FilterList from 'src/shared/components/FilterList'
+import InjectSecretOption from 'src/flows/pipes/RawFluxEditor/FluxInjectionOption'
+import {showOverlay, dismissOverlay} from 'src/overlays/actions/overlays'
+
+// Actions & Selectors
+import {getSecrets} from 'src/secrets/actions/thunks'
+import {getAllSecrets} from 'src/resources/selectors'
+
+// Context
+import {PipeContext} from 'src/flows/context/pipe'
+import {EditorContextType} from 'src/flows/context/editor'
+
+// Utils
+import {event} from 'src/cloud/utils/reporting'
+import {Secret} from 'src/types'
+
+const FilterSecrets = FilterList<Secret>()
+
+const SecretToSelect: FC<{
+  secret: Secret
+  onClick: (secret: Secret) => void
+}> = ({secret, onClick}) => {
+  return (
+    <InjectSecretOption
+      option={secret}
+      onClick={onClick}
+      key={secret.id}
+      testID={`select-secret-${secret.id}`}
+    />
+  )
+}
+
+type ReduxProps = ConnectedProps<typeof connector>
+
+interface Props extends ReduxProps {
+  context: EditorContextType
+}
+
+const SecretsList: FC<Props> = ({
+  context: editorContext,
+  onDismissOverlay,
+  onShowOverlay,
+}) => {
+  const [searchTerm, setSearchTerm] = useState<string>('')
+  const dispatch = useDispatch()
+  const secrets = useSelector(getAllSecrets)
+  const {data} = useContext(PipeContext)
+  const {editor, calcInsertPosition, updateText} = editorContext
+  const {queries, activeQuery} = data || {queries: []}
+  const query = queries[activeQuery]
+
+  const handleCreateSecret = () => {
+    event('Create Secret Modal Opened')
+    onShowOverlay('create-secret', null, onDismissOverlay)
+  }
+
+  const inject = useCallback(
```

Comment:
is there enough overlap between this and the function injection to warrant some extra abstraction to merge those two mechanisms?

Reply:
I was thinking the same thing. Making a generalized "inject `onOwnLine` versus `withinSameLine`" logic choice, and then all the text creation etc is pulled into the EditorContext functions.

But then the lose the ability to do bespoke changes.

Instead, I may create an inject() method in the EditorContext, which refers the helper functions (calcPosition and updateText). Then based on the option based to this new inject() function, it does `onOwnLine` versus `withinSameLine`.

Reply:
done